internal/vault/client_factory.go

```diff
@@ -97,8 +97,11 @@ type cachingClientFactory struct {
 	clientCallbacks        []ClientCallbackHandler
 	callbackHandlerCh      chan Client
 	mu                     sync.RWMutex
+	muLocks                sync.RWMutex
+	encClientLock          sync.RWMutex
 	onceDoWatcher          sync.Once
 	callbackHandlerCancel  context.CancelFunc
+	cacheKeyLocks          map[ClientCacheKey]*sync.RWMutex
```

Comment:
Is this essentially a lock per client? Maybe naming this something like `clientLocks` would be more clear?

Reply:
Updated in d7d1b288c2c22332543c51175171f1f4ab9f4115